app/assets/javascripts/codeblocks.js

```diff
@@ -0,0 +1,24 @@
+$(() => {
+  $(".post--content pre > code").each((_, block) => {
+    const wrapper = block.parentElement;
+
+    const wrapperDiv = document.createElement('div');
+    wrapperDiv.style.position = 'relative';
+
+    const copyButton = document.createElement('button');
+    copyButton.style.position = 'absolute';
+    copyButton.style.right = 0;
+    copyButton.classList.add('copy-button', 'button', 'is-muted', 'is-outlined', 'has-margin-2');
+    copyButton.innerText = 'Copy';
+    copyButton.addEventListener('click', _ => {
+      navigator.clipboard.writeText(block.innerText);
+      copyButton.innerText = 'Copied!';
+      setTimeout(() => {
+        copyButton.innerText = 'Copy';
+      }, 2000);
+    });
+
+    wrapper.insertAdjacentElement('beforebegin', wrapperDiv);
+    wrapperDiv.append(copyButton, wrapper);
+  });
+});
```

Comment:
suggestion: I'd pick one - either use jQuery, or go vanilla - mixing both feels weird. This feels like it could be simpler with jQuery.